usaspending_api/common/helpers/decorators.py

```diff
@@ -0,0 +1,72 @@
+import logging
+
+from django.conf import settings
+from django.db import connection
+from django.db.utils import OperationalError
+
+from usaspending_api.common.exceptions import EndpointTimeoutException
+
+logger = logging.getLogger(__name__)
+
+
+def set_db_timeout(timeout_in_seconds=settings.DEFAULT_DB_TIMEOUT_IN_SECONDS):
+    """ Decorator used to set the database statement timeout within the Django app scope
+
+        Args:
+            timeout_in_seconds: timeout value to set, in milliseconds
+
+        NOTE:
+            The statement_timeout is only set for this specific connection. Any subsequent request will have a default
+            value of 0. The alteration DOES NOT persist across connections and since each request is a new connection to
```

Comment:
This seems to contradict the finally, where you do set it back to 0. Maybe just reword, and say "while this won't persist across connections, we set it back to 0 as a precaution after the decorated function completes", in case anything reuses that connection.